packages/runtime/datastore/src/fluidHandle.ts

```diff
@@ -36,40 +49,41 @@ export class FluidObjectHandle<T extends FluidObject = IFluidObject> implements
         public readonly routeContext: IFluidHandleContext,
     ) {
         this.absolutePath = generateHandleContextPath(path, this.routeContext);
+        if (this.isAttached) {
```

Comment:
Tracking of visibility state (or graph attach state before) in handles is bound to have problems. There can be many handles to a data store or DDS.
For data stores, only the one stored in another DDS will have this state correctly updated. Other handles, if they were created before data store was visible (or attached) will not know this state.
For DDS, only the default handle will know this state. Any additional handles to it won't know this state.

It `bind` is called on these handles with another handle, the second one may not be bound even if the underlying object is locally visible.

Reply:
Also, since bind, attachGraph and isAttached are part of IFluidHandle, it defeats our model of using handles as light weight object for passing around within a container. For that purpose, handles don't need these APIs / properties and it only adds confusion.

These states are only ever interesting for the default DDS handle because it can bind other handles.  For other handles, like blob handle, these don't make sense. Tracking these states and `boundHandles` for data store handles don't make sense either because data store runtime already has these. So, this state is duplicated to all handles of a data store which makes its very hard to keep them in sync. Nobody should call bind on a data store's handle.

We should make a special purpose handle for DDSes and remove these from IFluidHandle.

Reply:
@vladsud @anthony-murphy